plugin/src/main/kotlin/me/tylerbwong/gradle/metalava/task/MetalavaSignature.kt

```diff
@@ -31,30 +31,28 @@ internal object MetalavaSignature : MetalavaTaskContainer() {
                 inputs.files(sources)
                 outputs.file(filename)
 
-                doFirst {
-                    val fullClasspath = (module.bootClasspath + module.compileClasspath).joinToString(File.pathSeparator)
+                val fullClasspath = (module.bootClasspath + module.compileClasspath).joinToString(File.pathSeparator)
```

Comment:
I believe I added the `doFirst` block so that the boot and compile classpaths wouldn't be accessed until after task configuration. I'll have to double check that removing `doFirst` doesn't cause any issues.

Reply:
So after looking into it a bit more, I think that instead of removing all `doFirst` blocks, we should instead add `extension.releaseType` as an input to the `metalavaCheckCompatibility` task. That way it is very obvious which inputs determine when this task should be re-run. We should be able to add something like `inputs.property("releaseType", extension.releaseType)` up above where we set the other task inputs.